include/otf2xx/reference_generator.hpp

```diff
@@ -150,7 +155,8 @@ class trace_reference_generator
     template <typename Definition>
     typename Definition::reference_type next()
     {
-        return get_generator<Definition>().next();
+        // TMP-code-obfuscator was here
+        return get_generator<Definition>().template next<typename Definition::reference_type>();
```

Comment:
To make this less ugly you can simply use `return static_cast<reference<Definition>>(get_generator<Definition>().next());`

The template and check in next is not really required. One should always use `trace_reference_generator` but not the raw `reference_generator`. And `get_generator` here makes already sure, that the references are convertible.